WolvenKit.App/ViewModels/HomePage/Pages/ModsViewModel.cs

```diff
@@ -0,0 +1,279 @@
+        private readonly AppViewModel _mainViewModel;
+
+        private readonly JsonSerializerOptions _options = new JsonSerializerOptions
+        {
+            WriteIndented = true,
+            DefaultIgnoreCondition = System.Text.Json.Serialization.JsonIgnoreCondition.WhenWritingNull,
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+            IgnoreReadOnlyProperties = true,
+        };
+
+        public ModsViewModel()
+        {
+            _settings = Locator.Current.GetService<ISettingsManager>();
+            _logger = Locator.Current.GetService<ILoggerService>();
+            _projectManager = Locator.Current.GetService<IProjectManager>();
+            _pluginService = Locator.Current.GetService<IPluginService>();
+
+            _mainViewModel = Locator.Current.GetService<AppViewModel>();
+
+            RefreshCommand = ReactiveCommand.Create(() => Refresh());
+            DeployCommand = ReactiveCommand.Create(() => Deploy());
+            LaunchGameCommand = ReactiveCommand.Create(() => LaunchGame());
+            CheckRedModCommand = ReactiveCommand.Create(() => CheckRedMod());
+            OpenModFolderCommand = ReactiveCommand.Create(() => OpenModFolder());
+
+            RemoveCommand = ReactiveCommand.Create(() => Remove());
+
+            LoadMods();
+        }
+
+        [Reactive] public ObservableCollection<ModInfoViewModel> Mods { get; set; } = new();
+        [Reactive] public ModInfoViewModel SelectedMod { get; set; }
+        [Reactive] public IEnumerable<ModInfoViewModel> SelectedMods { get; set; }
+        [Reactive] public bool IsProcessing { get; set; }
+
+        #region commands
+
+        public ICommand DeployCommand { get; private set; }
+        private async Task Deploy() => await DeployRedmod();
+        private IEnumerable<ModInfoViewModel> GetEnabledMods() => Mods.Where(x => x.IsEnabled);
+        private async Task<bool> DeployRedmod()
+        {
+            if (!_pluginService.IsInstalled(EPlugin.redmod))
+            {
+                _logger.Error("Redmod needs to be installed to deploy mods.");
+                switch (await Interactions.ShowMessageBoxAsync("The RedMod tools are not installed. Would you like to install them?", "RedMod not found"))
+                {
+                    case WMessageBoxResult.OK:
+                    case WMessageBoxResult.Yes:
+                        _homePageViewModel.NavigateTo(EHomePage.Plugins);
+                        break;
+                }
+
+                return false;
+            }
+
+            var result = false;
+            // compile with redmod
+            var redmodPath = Path.Combine(_settings.GetRED4GameRootDir(), "tools", "redmod", "bin", "redmod.exe");
+            if (!File.Exists(redmodPath))
+            {
+                _logger.Error("RedMod tools are not installed. Please go to WolvenKit plugins and install RedMod.");
+                result = await Task.FromResult(false);
```

Comment:
Same as the previous comment about `Task.FromResult` and friends.

I also see that you already check `!result`, could directly return there.